pkg/lib/cockpit.d.ts

```diff
@@ -77,7 +77,7 @@ declare module 'cockpit' {
         (event: CustomEvent<Parameters<E>>, ...args: Parameters<E>) => void;
 
     interface EventSource<EM extends EventMap> {
-        addEventListener<E extends keyof EM>(event: E, listener: EventListener<EM[E]>): void;
+        addEventListener<E extends keyof EM>(event: E, listener: EventListener<EM[E]>, options?: JsonObject): void;
```

Comment:
I think this is wrong. Our `addEventListener` implementation does not support the `options` parameter. Calling it with one should be an error.

Reply:
> It needs to store the result of addEventListener(), and explicitly call removeEventListener() on them on close.

Aren't event handlers removed by the GC when the channel object is collected? We don't remove event handlers from our channel objects explicitly in general.

If you want a guarantee that your handler is only called once, I think you are set as well: "ready" and "close" are only emitted once.

Reply:
> Aren't event handlers removed by the GC when the channel object is collected?

Perhaps/I hope so, I'm just never sure when exactly that happens. It's not synchronous like in C, Vala or Rust, AFAIK there's an actual GC running (similar to Python). This doesn't matter for this case, but it seems easy to have event handlers like systemd signals or D-Bus "PropertiesChanged" which wreak havoc when they happen multiple  times and call setState().